lib/models/__tests__/organization.test.ts

```diff
@@ -0,0 +1,202 @@
+    _self: "xyz",
+    _ts: "xyz",
+    organizationId: organizationId,
+    name: "MyOrganization",
+    id: "xyz",
+    kind: "IRetrievedOrganization",
+    version: toNonNegativeNumber(0).get,
+};
+
+describe("findOneOrganizationById", () => {
+
+    it("should resolve a promise to an existing organization", async () => {
+        const iteratorMock = {
+            executeNext: jest.fn((cb) => cb(undefined, ["result"], undefined)),
+        };
+
+        const clientMock = {
+            queryDocuments: jest.fn((__, ___) => iteratorMock),
+        };
+
+        const model = new OrganizationModel((clientMock as any) as DocumentDb.DocumentClient, organizationsCollectionUrl);
+
+        const result = await model.findByOrganizationId("id");
+
+        expect(result.isRight).toBeTruthy();
+        if (result.isRight) {
+            expect(result.right.isDefined).toBeTruthy();
+            expect(result.right.get).toEqual("result");
+        }
+    });
+
+    it("should resolve a promise to undefined if no organization is found", async () => {
+        const iteratorMock = {
+            executeNext: jest.fn((cb) => cb(undefined, [], undefined)),
+        };
+
+        const clientMock = {
+            queryDocuments: jest.fn((__, ___) => iteratorMock),
+        };
+
+        const model = new OrganizationModel((clientMock as any) as DocumentDb.DocumentClient, organizationsCollectionUrl);
+
+        const result = await model.findByOrganizationId("id");
+
+        expect(result.isRight).toBeTruthy();
+        if (result.isRight) {
+            expect(result.right.isEmpty).toBeTruthy();
+        }
+    });
+
+});
+
+describe("createOrganization", () => {
+
+    it("should create a new organization", async () => {
+        const clientMock: any = {
+            createDocument: jest.fn((_, newDocument, __, cb) => {
+                cb(undefined, {
+                    ...newDocument,
+                });
+            }),
+        };
+
+        const model = new OrganizationModel(clientMock, organizationsCollectionUrl);
+
+        const newOrganization: IOrganization = {
+            organizationId: organizationId,
+            name: "MyOrganization"
+        };
+
+        const result = await model.create(newOrganization, newOrganization.organizationId);
+
+        expect(clientMock.createDocument).toHaveBeenCalledTimes(1);
+        expect(clientMock.createDocument.mock.calls[0][1].kind).toBeUndefined();
+        expect(clientMock.createDocument.mock.calls[0][2]).toHaveProperty("partitionKey", organizationId);
+        expect(result.isRight).toBeTruthy();
+        if (result.isRight) {
+            expect(result.right.organizationId).toEqual(newOrganization.organizationId);
+            expect(result.right.id).toEqual(`${organizationId}-${"0".repeat(16)}`);
+            expect(result.right.version).toEqual(0);
+        }
+    });
+
+    it("should reject the promise in case of error", async () => {
```

Comment:
also here the promise is actually resolved to an error value (a `Left<DocumentDb.QueryError>`)